src/Common/SharedAccessSignatureHelper.php

```diff
@@ -64,6 +64,156 @@ public function __construct($accountName, $accountKey)
+     *                                    - {cont}/{blob}: for blobs
+     *                                    - {share}/{file}: for files
+     * @param  string $signedExpiry       Signed expiry date.
+     * @param  string $signedStart        Signed start date.
+     * @param  string $signedIdentifier   Signed identifier.
+     * @param  string $signedIP           Signed IP address.
+     * @param  string $signedProtocol     Signed protocol.
+     * @param  string $cacheControl       Cache-Control header (rscc).
+     * @param  string $contentDisposition Content-Disposition header (rscd).
+     * @param  string $contentEncoding    Content-Encoding header (rsce).
+     * @param  string $contentLanguage    Content-Language header (rscl).
+     * @param  string $contentType        Content-Type header (rsct).
+     *
+     * @return string                     Service Shared Access Signature.
+     */
+    public function generateServiceSharedAccessSignatureToken(
+        $signedPermissions,
+        $signedService,
+        $signedResource,
+        $signedExpiry,
+        $signedStart = "",
+        $signedIdentifier = "",
+        $signedIP = "",
+        $signedProtocol = "",
+        $cacheControl = "",
+        $contentDisposition = "",
+        $contentEncoding = "",
+        $contentLanguage = "",
+        $contentType = "",
+        $startingPartitionKey = "",
+        $startingRowKey = "",
+        $endingPartitionKey = "",
+        $endingRowKey = ""
+    ) {
+        // use the latest version
+        $signedVersion = Resources::STORAGE_API_LATEST_VERSION;
+
+        // validate and sanitize signed service
+        $signedService = $this->validateAndSanitizeSignedService($signedService);
+        // for a service SAS, the signed service should be limited to one
+        Validate::isTrue(
+            (strlen($signedService) === 1),
+            sprintf(Resources::INVALID_STRING_LENGTH, 'signedService', '1')
+        );
+
+        // validate and sanitize signed permissions
+        $signedPermissions = $this->validateAndSanitizeSignedPermissions($signedPermissions);
+
+        // check that expiracy is valid
+        Validate::isString($signedExpiry, 'signedExpiry');
+        Validate::notNullOrEmpty($signedExpiry, 'signedExpiry');
+        Validate::isDateString($signedExpiry, 'signedExpiry');
+
+        // check that signed start is valid
+        Validate::isString($signedStart, 'signedStart');
+        if (strlen($signedStart) > 0) {
+            Validate::isDateString($signedStart, 'signedStart');
+        }
+
+        // check that signed IP is valid
+        Validate::isString($signedIP, 'signedIP');
+
+        // validate and sanitize signed protocol
+        $signedProtocol = $this->validateAndSanitizeSignedProtocol($signedProtocol);
+
+        // check that signed identifier is valid
+        Validate::isString($signedIdentifier, 'signedIdentifier');
+        Validate::isTrue(
```

Comment:
If the signedIdentifier is provided, some other parameter is not allowed - start, expiry, permission.